tests/UnitTests/Rsp.IrasService.UnitTests/Web/Controllers/ApplicationsControllerTests/GetPaginatedApplicationsByRespondent.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Rsp.IrasService.Application.CQRS.Queries;
+using Rsp.IrasService.Application.DTOS.Responses;
+using Rsp.IrasService.WebApi.Controllers;
+
+namespace Rsp.IrasService.UnitTests.Web.Controllers.ApplicationsControllerTests;
+
+public class GetPaginatedApplicationsByRespondent : TestServiceBase
+{
+    private readonly ApplicationsController _controller;
+
+    public GetPaginatedApplicationsByRespondent()
+    {
+        _controller = Mocker.CreateInstance<ApplicationsController>();
+    }
+
+    [Theory]
+    [AutoData]
+    public async Task GetPaginatedApplicationsByRespondent_ShouldReturnApplications_WhenDataExists(string respondentId,
+        PaginatedResponse<ApplicationResponse> mockResponse)
+    {
+        // Arrange
+        var mockMediator = Mocker.GetMock<IMediator>();
+        mockMediator
+            .Setup(m => m.Send(
+                It.Is<GetPaginatedApplicationsWithRespondentQuery>(q =>
+                    q.RespondentId == respondentId &&
+                    q.SearchQuery == null &&
+                    q.PageIndex == 1 &&
+         q.PageSize == 5),
+                It.IsAny<CancellationToken>()))
+            .ReturnsAsync(mockResponse);
```

Comment:
use the following style when composing a long setup:

```suggestion
        mockMediator
            .Setup
            (
                m => m.Send
                (
                    It.Is<GetPaginatedApplicationsWithRespondentQuery>
                    (
                        q =>
                            q.RespondentId == respondentId &&
                            q.SearchQuery == null &&
                            q.PageIndex == 1 &&
                            q.PageSize == 5
                    ),
                    It.IsAny<CancellationToken>()
                )
            )
            .ReturnsAsync(mockResponse);
```